src/ThriftSQL/HiveQuery.php

```diff
@@ -59,20 +67,31 @@ public function fetch( $maxRows ) {
       throw new \ThriftSQL\Exception( "Query is not ready. Call `->wait()` before `->fetch()`" );
     }
     try {
-      $TFetchResultsResp = $this->_client->FetchResults( new \ThriftSQL\TFetchResultsReq( array(
+      $TFetchResultsResp = $this->_client->FetchResults( new \ThriftGenerated\TFetchResultsReq( array(
         'operationHandle' => $this->_resp->operationHandle,
         'maxRows' => $maxRows,
       ) ) );
       /**
        * NOTE: $TFetchResultsResp->hasMoreRows appears broken, it's always
        * false so one needs to keep fetching until they run out of data.
        */
-      if ( $TFetchResultsResp->results instanceof \ThriftSQL\TRowSet && !empty( $TFetchResultsResp->results->columns ) ) {
+      if ( $TFetchResultsResp->results instanceof \ThriftGenerated\TRowSet && !empty( $TFetchResultsResp->results->columns ) ) {
         return $this->_colsToRows( $TFetchResultsResp->results->columns );
       }
       return array();
     } catch ( Exception $e ) {
-      throw new \ThriftSQL\Exception( $e->getMessage() );
+      throw new \ThriftSQL\Exception( $e->getMessage(), $e->getCode(), $e );
+    }
+  }
+
+  public function close() {
+    try {
```

Comment:
should we wrap all exceptions in a `\ThriftSQL\Exception`?

Reply:
So I'm just catching and ignoring with a `finally` only the Impala service documents that it can throw an exception on close but feels like we can safely ignore any potential issues as there's no way to actually recover from not being able to close a query handle. (Because that just means something is totally broken and the query handle is invalid anyway.)